examples/common/text_data.py

```diff
@@ -137,13 +137,65 @@ def __getitem__(self, idx: int) -> Dict[str, Any]:
         return token_sample
 
 
+class ConcatenatedSequenceCollatorWrapper:
+    """Collator wrapper to add sequence_id to batch."""
+
+    def __init__(self,
+                 base_collator: Callable,
+                 eos_token_id: Optional[int] = None,
+                 bos_token_id: Optional[int] = None):
+        self.base_collator = base_collator
+        if (eos_token_id is None) and (bos_token_id is None):
+            raise ValueError(
+                'Must supply a value for either eos_token_id or bos_token_id, but got None for both.'
+            )
+        if (eos_token_id is not None) and (bos_token_id is not None):
+            raise ValueError(
+                'Must supply a value for either eos_token_id or bos_token_id, but both were given.'
+            )
+        if eos_token_id is None:
+            self.split_token_id = bos_token_id
+            self.eos_mode = False
+        else:
+            self.split_token_id = eos_token_id
+            self.eos_mode = True
+        assert self.split_token_id is not None
+
+    def __call__(self, examples: List[Any]) -> Dict[str, torch.Tensor]:
+        batch = self.base_collator(examples)
+        if self.eos_mode:
+            return self.sequence_id_from_eos(batch, self.split_token_id)
+        else:
+            return self.sequence_id_from_bos(batch, self.split_token_id)
+
+    @staticmethod
+    def sequence_id_from_eos(batch: Dict[str, torch.Tensor],
+                             split_token_id: int) -> Dict[str, torch.Tensor]:
```

Comment:
does this need to be a staticmethod?
if not, then you don't need to pass `split_token_id` you can just use `self.eos_token_id`
(assuming you use my previous suggestion)

same comment for `sequence_id_from_bos`

Reply:
It doesn't need to be, but being a staticmethod makes testing a bit more convenient. I'll clean this part of the code up a bit and add some tests to actually take advantage of that choice, haha.